src/Routing/RouteSubscriber.php

```diff
@@ -18,6 +19,14 @@ protected function alterRoutes(RouteCollection $collection) {
     if ($route = $collection->get('entity.media.add_page')) {
       $route->setDefault('_controller', MediaAdd::class . '::addPage');
     }
+    if ($route = $collection->get('editor.media_dialog')) {
+      $route->setDefault('_form', StanfordMediaDialogForm::class);
+    }
+    if ($route = $collection->get('dropzonejs.upload')) {
+      $requirements = $route->getRequirements();
+      unset($requirements['_csrf_token']);
```

Comment:
Tell me more about why you are removing the CSRF token?

Reply:
it just doesnt work from the wysiwyg. I think its not getting the correct context from the page or something along those lines. And its an internal thing. I can't modify it without altering core entirely.

Reply:
you can verify by commenting out that line, clearing cache and trying to upload an image through the wysiwyg

Reply:
Aye, you might have the generate that token with your form override as it is used to ensure the proper session and submission.

Reply:
I read that you're not supposed to manage the token yourself. The routing system handles it

Reply:
> Note that, in order for the token to be added, the link must be generated using the url_generator service via route name ("example" here) rather than as a manually constructed path.

From https://www.drupal.org/docs/8/api/routing-system/access-checking-on-routes/csrf-access-checking

Reply:
I haven't dug in to the inner workings of the csrf handling so I am unsure about the path forward but I am uncomfortable with removing it altogether as it feels like a security concern.

Reply:
> CSRF protection is now integrated into the routing access system and should be used for any URLs that perform actions or operations that do not use a form callback

Are you replacing something that was not a form with a form callback?

Reply:
not that i'm aware of. I'm replacing a traditional file upload field. thats all. but the CSRF token has issues when dropping a file into the area. before any button or callback is performed